async_generator/_tests/test_async_generator.py

```diff
@@ -779,3 +849,170 @@ def test_refcnt():
     for msg in recwarn:  # pragma: no cover
         print(msg)
         assert not issubclass(msg.category, RuntimeWarning)
+
+
+################################################################
+#
+# GC hooks
+#
+################################################################
+
+
+@pytest.fixture
+def local_asyncgen_hooks():
+    old_hooks = get_asyncgen_hooks()
+    yield
+    set_asyncgen_hooks(*old_hooks)
+
+
+def test_gc_hooks_interface(local_asyncgen_hooks):
+    def one(agen):  # pragma: no cover
+        pass
+
+    def two(agen):  # pragma: no cover
+        pass
+
+    set_asyncgen_hooks(None, None)
+    assert get_asyncgen_hooks() == (None, None)
+    set_asyncgen_hooks(finalizer=two)
+    assert get_asyncgen_hooks() == (None, two)
+    set_asyncgen_hooks(firstiter=one)
+    assert get_asyncgen_hooks() == (one, two)
+    set_asyncgen_hooks(finalizer=None, firstiter=two)
+    assert get_asyncgen_hooks() == (two, None)
+    set_asyncgen_hooks(None, one)
+    assert get_asyncgen_hooks() == (None, one)
+    tup = (one, two)
+    set_asyncgen_hooks(*tup)
+    assert get_asyncgen_hooks() == tup
+
+    with pytest.raises(TypeError):
+        set_asyncgen_hooks(firstiter=42)
+
+    with pytest.raises(TypeError):
+        set_asyncgen_hooks(finalizer=False)
+
+    def in_thread(results=[]):
```

Comment:
nitpick: we always pass `results` to `in_thread`, so it should be a mandatory argument (no default)

Reply:
done (and using a list as a default argument, too - what was I even thinking?)